module-4/backend/week_two.md

```diff
@@ -5,18 +5,30 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's one difference between ES5 and ES6?
+    * Classes are available in ES6 which are more reminiscent of other object oriented languages such as Ruby. 
 2. What's the difference between asynchronous and synchronous JavaScript? 
+    * Synchonous Javascript executes line by line while asychronous allows other tasks to be completed while the rest of the JS is running.
 3. What are the four pillars of Object Oriented programming?
+    * Encapsulation, Abstraction, Inheritance, Polymorphism
 4. What are some tools available in JavaScript to help you write object oriented code?
+    * Classes, Constructors, Prototypes
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+    * Single Responsiblity and the abiltiy to unit test. Also organization is important for finding specific functionality.
 6. What's a callback function and what are some reasons when we use/need callback functions?
+    * Callback functions occur when an event triggers it. We need them on event listeners to execute tasks once that event happens. 
```

Comment:
They also exist anytime a function is passed to another function to be invoked by that 2nd function.